syntax = "proto3";

package Blocksettle.ArmoryEvents;

enum EventType
{
   NewBlockEventType = 0;
   ZCEventType = 1;
};

message EventHeader
{
   EventType   event_type = 1;
   bytes       event_data = 2;
};

message NewBlockEvent
{
   uint32      height = 1;
};

message ZCEvent
{
   message TXEntry {
      bytes   tx_hash = 1;
      string  wallet_id = 2;
      int64   value = 3;
      uint32  block_num = 4;
      uint32  time = 5;
      bool    rbf = 6;
      bool    chained_ZC = 7;
   }
   repeated TXEntry zc_entries = 1;
};
